tests/test_find_django.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Tests for find annotations in Django models.
+"""
+from mock import DEFAULT, patch
+
+from code_annotations.django_reporting_helpers import get_model_id
+from tests.fake_models import (
+    FakeBaseModelNoAnnotation,
+    FakeBaseModelWithAnnotation,
+    FakeBaseModelWithNoDocstring,
+    FakeChildModelMultipleAnnotation,
+    FakeChildModelMultipleWithBrokenAnnotations,
+    FakeChildModelSingleAnnotation,
+    FakeChildModelSingleWithAnnotation
+)
+from tests.helpers import EXIT_CODE_FAILURE, EXIT_CODE_SUCCESS, call_script_isolated
+
+
+@patch.multiple(
+    'code_annotations.find_django',
+    get_models_requiring_annotations=DEFAULT
+)
+def test_find_django_simple_success(**kwargs):
+    """
+    Tests the basic case where all models have annotations, with an empty safelist.
+    """
+    test_models = {FakeChildModelSingleAnnotation, FakeChildModelMultipleAnnotation, FakeChildModelSingleWithAnnotation}
+    mock_get_models_requiring_annotations = kwargs['get_models_requiring_annotations']
+    mock_get_models_requiring_annotations.return_value = (
+        test_models,
+        set()
+    )
+
+    def report_callback(report_contents):
+        """
+        Get the text of the report and make sure all of the expected models are in it.
+
+        :param report_contents: Raw text contents of the generated report file
+        """
+        for model in test_models:
+            assert 'object_id: {}'.format(get_model_id(model)) in report_contents
+
+    result = call_script_isolated(
+        ['django_find_annotations', '--config_file', 'test_config.yml', '--lint', '--report'],
+        test_filesystem_report_cb=report_callback
+    )
+
+    assert result.exit_code == EXIT_CODE_SUCCESS
+    assert 'Linting passed without errors' in result.output
+    assert 'Generating report to' in result.output
+    assert 'Search found 5 annotations' in result.output
+
+
+@patch.multiple(
+    'code_annotations.find_django',
+    get_models_requiring_annotations=DEFAULT,
```

Comment:
Does this trailing comma have any effect?

Reply:
Nope, it's just left over from some intermediate changes.